onnxruntime/test/onnx/runner.cc

```diff
@@ -461,12 +480,16 @@ EXECUTE_RESULT DataRunner::RunTaskImpl(size_t task_id) {
   return res;
 }
 
-void SeqTestRunner::Start(ORT_CALLBACK_INSTANCE pci, size_t) {
-  const size_t data_count = c_.GetDataCount();
-  for (size_t idx_repeat = 0; idx_repeat != repeat_count_; ++idx_repeat) {
-    for (size_t idx_data = 0; idx_data != data_count; ++idx_data) {
-      RunTask(idx_data, nullptr);
+void SeqTestRunner::Start(ORT_CALLBACK_INSTANCE pci, size_t) noexcept {
+  try {
+    const size_t data_count = c_.GetDataCount();
+    for (size_t idx_repeat = 0; idx_repeat != repeat_count_; ++idx_repeat) {
+      for (size_t idx_data = 0; idx_data != data_count; ++idx_data) {
+        RunTask(idx_data, nullptr);
+      }
     }
+  } catch (...) {
```

Comment:
In addition to this, should we catch std::exception as well so that we can log the reason for the failure?

Reply:
RunTask handles test case failures. This catch here is largely a pre-caution.

---
In reply to: [452673014](https://github.com/microsoft/onnxruntime/pull/4475#discussion_r452673014) [](ancestors = 452673014)

Reply:
Would be nice to add a comment. It's not obvious.

Reply:
we need to re-write

---
In reply to: [453008098](https://github.com/microsoft/onnxruntime/pull/4475#discussion_r453008098) [](ancestors = 453008098)